common/scala/src/main/scala/org/apache/openwhisk/core/entity/ActivationResult.scala

```diff
@@ -86,19 +88,25 @@ protected[core] object ActivationResponse extends DefaultJsonProtocol {
     }
   }
 
-  private def error(code: Int, errorValue: JsValue) = {
+  private def error(code: Int, errorValue: JsValue, size: Option[Int]) = {
     require(code == ApplicationError || code == DeveloperError || code == WhiskError)
-    ActivationResponse(code, Some(JsObject(ERROR_FIELD -> errorValue)))
+    ActivationResponse(code, Some(JsObject(ERROR_FIELD -> errorValue)), size)
   }
 
-  protected[core] def success(result: Option[JsValue] = None) = ActivationResponse(Success, result)
-
-  protected[core] def applicationError(errorValue: JsValue) = error(ApplicationError, errorValue)
-  protected[core] def applicationError(errorMsg: String) = error(ApplicationError, JsString(errorMsg))
-  protected[core] def developerError(errorValue: JsValue) = error(DeveloperError, errorValue)
-  protected[core] def developerError(errorMsg: String) = error(DeveloperError, JsString(errorMsg))
-  protected[core] def whiskError(errorValue: JsValue) = error(WhiskError, errorValue)
-  protected[core] def whiskError(errorMsg: String) = error(WhiskError, JsString(errorMsg))
+  protected[core] def success(result: Option[JsValue] = None, size: Option[Int] = None) =
```

Comment:
Why is size an option? Backward compatibility? If so, perhaps should be dealt with in deserializer. This would enforce the size always being Some value.

Reply:
Size is only applicable for case where error is based on some response (similar to via result is an Option). In some case we just construct a failed response due to connection error etc and there size is not applicable